# Genetic-Expression-Data-Processing-Visualization-Analysis-Workshop


## Workshop Information

This workshop is organized by the **Advancing Cervical Cancer Education and Screening in Sub Saharan Africa (ACCESS) venture team** from **Engineers Without Borders (EWB) at the University of British Columbia (UBC)**.

**Time**: 5-7 pm  
**Date**: 29 November  
**Location**: ORCH 3004  
**Duration**: 2 hours


---

## Workshop Requirements

This workshop uses **Python**, a free and versatile programming language compatible with major operating systems, including macOS, Linux, and Windows. We will work specifically with **Python 3**.

To develop and run our Python scripts, we will use **Jupyter Lab**, a user-friendly computing environment. Please ensure that you use **Jupyter Lab** and not **Jupyter Notebook** for this course.

While it is possible to install Python and Jupyter Lab separately, this is not recommended. The preferred approach is to install **Anaconda**, which includes:

- Python and its libraries
- Jupyter Lab
- Additional tools like Spyder for a user-friendly Python programming experience

You can download Anaconda [here](https://www.anaconda.com/download).

Additionally, we will use the following Python libraries in this workshop:
- **scikit-learn**
- **SciPy**
- **pandas**
- **NumPy**

These libraries are typically included in the Anaconda distribution. If they are not available, you can use a package manager to install them.

---

## Directory Structure (Good Practice)

It is a good practice to organize your project into appropriate directories. For this workshop, the following directory structure is recommended:


- **`project/`**  
  The root of the project directory.

  - **`src/`**  
    Source code files (your Python scripts).
  
  - **`data/`**  
    Raw datasets.
  
  - **`results/`**  
    Generated results, such as saved figures or outputs.
    
This structure ensures that your project remains organized and easy to navigate, keeping your code, data, and results separate.

---

## Dataset Overview

We will be utilizing the **Cervical Squamous Cell Carcinoma** dataset provided by the **Cancer Genome Project**. The dataset can be accessed here: [Dataset Link](https://bit.ly/3YVssrK).

This dataset contains:
- Clinical data
- Genomic (DNA-sequencing) data
- Expression (RNA-sequencing) data

For today's workshop, we will only use the **clinical** and **expression** data. Please download the dataset and place it inside the **data/** directory.

---


